src/main/java/com/google/idea/perf/TracerController.kt

```diff
@@ -86,20 +86,35 @@ class TracerController(
     }
 
     private fun dataRefreshLoop() {
+        val startTime = System.nanoTime()
+        var shouldUpdateTreeUi = false
+
         val treeDeltas = CallTreeManager.collectAndReset()
         if (treeDeltas.isNotEmpty()) {
             treeDeltas.forEach(callTree::accumulate)
-            updateUi()
+            shouldUpdateTreeUi = true
         }
+
+        val endTime = System.nanoTime()
+        val elapsedNanos = endTime - startTime
+        updateUi(shouldUpdateTreeUi, elapsedNanos)
     }
 
     /** Refreshes the UI with the current state of [callTree]. */
-    private fun updateUi() {
+    private fun updateUi(shouldUpdateCallTree: Boolean = true, elapsedTime: Long? = null) {
         val allStats = TreeAlgorithms.computeFlatTracepointStats(callTree)
         val visibleStats = allStats.filter { it.tracepoint != Tracepoint.ROOT }
+
         // We use invokeAndWait to ensure proper backpressure for the data refresh loop.
         getApplication().invokeAndWait {
-            view.listView.setTracepointStats(visibleStats)
+            if (shouldUpdateCallTree) {
+                view.listView.setTracepointStats(visibleStats)
```

Comment:
Updating the UI here could be expensive, so I think it makes sense to include its overhead in the calculation of "refresh time".

One idea is to compute elapsed time at the end of `dataRefreshLoop` and use a separate `invokeAndWait` call to update just the refresh time label. (In that case we could consider renaming `updateUi` to `updateTreeUi`).